React/Views/RCTTabBar.m

```diff
@@ -175,6 +175,18 @@ - (void)setItemPositioning:(UITabBarItemPositioning)itemPositioning
 
 #pragma mark - UITabBarControllerDelegate
 
+#if TARGET_OS_TV
+
+- (void)tabBarController:(UITabBarController *)tabBarController didSelectViewController:(nonnull UIViewController *)viewController
```

Comment:
`didSelectViewController` seems to work fine on iOS too... if desired, we could just use this method for both iOS and tvOS.